main.go

```diff
@@ -17,6 +23,95 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
+type MigrationRunner struct {
+	path string
+}
+
+func NewMigrationRunner(path string) *MigrationRunner {
+	return &MigrationRunner{path: path}
+}
+
+func (m *MigrationRunner) Run(ctx context.Context, conn *sql.Conn, lastRanId int) error {
+	files, err := m.loadMigrationFiles()
+	if err != nil {
+		return err
+	}
+
+	sort.Slice(files, func(i, j int) bool {
+		id1 := extractMigrationID(files[i])
+		id2 := extractMigrationID(files[j])
+
+		return id1 < id2
+	})
+
+	// var index int = 0
+	// for _, file := range files {
+	// 	if extractMigrationID(file) <= lastRanId {
+	// 		index++
+	// 	}
+	// }
+
+	fileSlice := files[lastRanId+1:]
+
+	for _, file := range fileSlice {
+		readFile, err := os.Open(file)
+
+		if err != nil {
+			fmt.Println(err)
+		}
+		fileScanner := bufio.NewScanner(readFile)
+
+		fileScanner.Split(bufio.ScanLines)
+		for fileScanner.Scan() {
+
+			if strings.Trim(fileScanner.Text(), "") == "" {
+				continue
+			}
+			if _, err := conn.ExecContext(ctx, fileScanner.Text()); err != nil {
+				return fmt.Errorf("failed to execute migration %s: %w", file, err)
+			}
+		}
```

Comment:
Reading each file line-by-line totally works, so don't feel obligated to change this.

If, however, you wanted to support multi-line commands in your migration files, e.g.
```sql
CREATE TABLE customers (
  ID INT AUTO_INCREMENT PRIMARY KEY, 
  email VARCHAR(255), 
  state VARCHAR(2)
);

CREATE TABLE products (
 ...
 ```
 
You could split this file up by semicolons `;` instead of by lines since that's what sql uses to separate commands.